M2/Macaulay2/packages/MacaulayPosets.m2

```diff
@@ -0,0 +1,1956 @@
+newPackage(
+    "MacaulayPosets",
+    Version => "1.0",
+    Date => "July 1, 2025",
+    Headline => "Macaulay posets",
+        Authors => { {Name => "Penelope Beall", Email => "pbeall@ucdavis.edu", HomePage=>"https://pbeall.github.io"}, {Name => "Yu Olivier Li", Email => "yol1@st-andrews.ac.uk"}  },
+    PackageExports => {
+        "Posets"
+    },
+    Keywords => {"Combinatorial Commutative Algebra"}
+)
+export {
+    "PosetMap",
+    "isRankPreserving",
+    "getMons",
+    "getPoset",
+    "upperShadow",
+    "lowerShadow",
+    "macaulayOrders",
+    "isMacaulay",
+    "isAdditive",
+    "posetWedgeProduct",
+    "posetFiberProduct",
+    "posetClosedProduct",
+    "posetConnectedSum",
+    "ringFiberProduct",
+    "ringConnectedSum",
+    "MaxDegree",
+    "MaxTime",
+    "TikZ",
+    "Visual",
+    "AllOrders"
+}
+needsPackage "Posets";
+needsPackage "Visualize";
```

Comment:
`needsPackage "Posets"` is redundant since it's already in `PackageExports` above, and it would be good to add `Visualize` there instead (or `PackageImports`, depending on whether you want users to also have access to its functionality or not).